src/TraceEvent/TraceEvent.cs

```diff
@@ -3145,6 +3145,16 @@ public sealed class TraceEventDispatcherOptions
     /// </summary>
     public abstract unsafe class TraceEventDispatcher : TraceEventSource
     {
+        /// <summary>
+        /// Obtains the correct TraceEventDispatcher for multiple ETW files.
+        /// </summary>
+        /// <param name="fileNames">A list of paths to trace files.</param>
+        /// <returns>A TraceEventDispatcher for the given trace files.</returns>
+        public static TraceEventDispatcher GetDispatcherFromMultiEventTraceLogFiles(IEnumerable<string> fileNames, TraceEventDispatcherOptions options = null)
```

Comment:
I don't think you should add this method, but instead just call `new ETWTraceEventSource` directly, and pass that to a new method in `TraceLog` that looks like this:

```
        public static string CreateFromEventTraceLogFile(TraceEventDispatcher source, string etlxFilePath, TraceLogOptions options = null)
        {
            if (etlxFilePath == null)
            {
                throw new ArgumentNullException(nameof(etlxFilePath));
            }

            if (source.EventsLost != 0 && options != null && options.OnLostEvents != null)
            {
                options.OnLostEvents(false, source.EventsLost, 0);
            }

            CreateFromTraceEventSource(source, etlxFilePath, options);

            return etlxFilePath;
        }

```

This ensures that if someone wants to do more custom things than what TraceEvent knows how to do, that they select the right `EventSource` class rather than using have to write a method that makes sure that all of the files are of the right type before creating the corresponding `EventSource`.  Also, most `EventSources` don't support multiple files.